pandas/core/internals/blocks.py

```diff
@@ -1344,6 +1353,10 @@ def delete(self, loc) -> list[Block]:
         new_blocks: list[Block] = []
 
         previous_loc = -1
+        # TODO(CoW): This is tricky, if parent block goes out of scope
+        # all split blocks are referencing each other even though they
+        # don't share data
+        refs = self.refs if self.refs.has_reference() else None
```

Comment:
Can we create kind of a "clone" of the `refs` (duplicate it)? Create a new `refs` that creates new weakrefs to the same objects. But that ensures that both lists are independent, and the two split blocks don't reference each other

Reply:
This is doable (in theory), but not here (would make things way more complicated). The thing that is not clear to me with this solution is how we would update all parent blocks that share this reference object
You could have:

a -> b -> c -> split happens ->d1, d2, d3, ...

When you clone the reference object, you would have to make sure that a, b and c are all aware of the new object. The simplest solution would probably be, that a reference object can point to another reference object which would live with the split blocks. These chains can get arbitrarily deep though, so we have to think the exact approach through.

Good think with this class is though, that we can hide the implementation inside it without having to worry in other areas.

Reply:
I'd prefer doing this in a follow up. We are calling ``_slice_take_blocks_ax0`` under the hood which would have to avoid tracking references via a keyword. Not super difficult but would make the pr messier than necessary. But will get a pr up before 2.0

Reply:
Yep, to be clear my "shorter term" was meant as "after this PR and if we don't directly have a better general solution for  2.0" ;)